internal_worker.go

```diff
@@ -765,13 +854,19 @@ var once sync.Once
 // Singleton to hold the host registration details.
 var thImpl *hostEnvImpl
 
+func newHostEnvironment() *hostEnvImpl {
+	return &hostEnvImpl{
+		workflowFuncMap:  make(map[string]interface{}),
+		workflowAliasMap: make(map[string]string),
+		activityFuncMap:  make(map[string]activity),
+		activityAliasMap: make(map[string]string),
+		encoding:         gobEncoding{},
+	}
+}
+
 func getHostEnvironment() *hostEnvImpl {
 	once.Do(func() {
-		thImpl = &hostEnvImpl{
-			workflowFuncMap: make(map[string]interface{}),
-			activityFuncMap: make(map[string]interface{}),
-			encoding:        gobEncoding{},
-		}
+		thImpl = newHostEnvironment()
```

Comment:
That would create a new host environment everytime instead of reusing the existing global one. Not the behavior we want.